src/phoenix/datasets/evaluators/utils.py

```diff
@@ -0,0 +1,164 @@
+import functools
+import inspect
+from typing import Any, Callable, Optional
+
+from phoenix.datasets.types import (
+    AnnotatorKind,
+    CanAsyncEvaluate,
+    CanEvaluate,
+    EvaluationResult,
+    EvaluatorCallable,
+    Example,
+    ExperimentEvaluator,
+    ExperimentRun,
+    JSONSerializable,
+)
+
+
+def _unwrap_json(obj: JSONSerializable) -> JSONSerializable:
+    if isinstance(obj, dict):
+        if len(obj) == 1:
+            key = next(iter(obj.keys()))
+            output = obj[key]
+            assert isinstance(
+                output, (dict, list, str, int, float, bool, type(None))
+            ), "Output must be JSON serializable"
+            return output
+    return obj
+
+
+def _validate_signature(sig: inspect.Signature) -> None:
+    # Check that the wrapped function has a valid signature for use as an evaluator
+    # If it does not, raise an error to exit early before running evaluations
+    params = sig.parameters
+    valid_named_params = {"experiment_input", "output", "reference", "metadata"}
+    if len(params) == 0:
+        raise ValueError("Evaluation function must have at least one parameter.")
+    if len(params) > 1:
+        not_found = set(params) - valid_named_params
+        if not_found:
+            raise ValueError(
+                (
+                    f"Invalid parameter names in evaluation function: {', '.join(not_found)}. "
+                    "Parameters names for multi-argument functions must be "
+                    f"any of: {', '.join(valid_named_params)}."
+                )
+            )
+
+
+def _bind_signature(
+    sig: inspect.Signature, example: Example, experiment_run: ExperimentRun
+) -> inspect.BoundArguments:
+    if experiment_run.output:
+        output = experiment_run.output.result
+    else:
+        output = None
+    parameter_mapping = {
+        "experiment_input": example.input,
+        "output": output,
+        "reference": example.output,
+        "metadata": example.metadata,
+    }
+    params = sig.parameters
+    if len(params) == 1:
+        parameter_name = next(iter(params))
+        if parameter_name in parameter_mapping:
+            return sig.bind(parameter_mapping[parameter_name])
+        else:
+            return sig.bind(parameter_mapping["output"])
+    else:
+        return sig.bind_partial(**{name: parameter_mapping[name] for name in params})
+
+
+def create_evaluator(
+    annotator: AnnotatorKind = AnnotatorKind.CODE,
```

Comment:
```suggestion
    kind: AnnotatorKind = AnnotatorKind.CODE,
```
Should also accept a string so you don't have to import.

Reply:
sounds good!